### DISCUSSION TITLE
Temporal theme modeling and theme evolution Sankey diagram

### DISCUSSION THREAD

**Discussion by superseanyoung** [MaartenGr/BERTopic#2239] [General]
I want to model the topic in time, then get the topic in different time periods, and calculate the cosine similarity between these topic vectors, but I don't know how to find a vector representation of a topic.

**Comment by MaartenGr** [comment]
That is currently not possible since these slices are not saved (they are quite big if you capture them all). That said, you might be able to extract them if you manually adjust some of the code here: 

https://github.com/MaartenGr/BERTopic/blob/84dbf36d1f2ecc1eb11ba453e39fc6e2dda8b5e3/bertopic/_bertopic.py#L874

You would have to save and return all the `c_tf_idf` matrices (note that the topic vectors are not re-calculated here, so they are not used).
